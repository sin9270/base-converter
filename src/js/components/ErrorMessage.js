'use strict';

import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  message: PropTypes.string.isRequired
};

const ErrorMessage = props => {
  return <span className="errorMassage">{props.message}</span>;
};

ErrorMessage.propTypes = propTypes;

export default ErrorMessage;
