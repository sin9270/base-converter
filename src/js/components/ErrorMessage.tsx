import * as React from 'react';

interface Props {
  message: string;
}

const ErrorMessage: React.FC<Props> = (props) => {
  return <span className="errorMassage">{props.message}</span>;
};

export default ErrorMessage;
