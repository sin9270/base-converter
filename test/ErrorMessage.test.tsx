/* eslint @typescript-eslint/no-explicit-any: 0 */
import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';

import ErrorMessage from '../src/js/components/ErrorMessage';

describe('test ErrorMessage', () => {
  it('propsの値を表示しているかチェック', () => {
    const wrapper = shallow(<ErrorMessage message="An error occurred!" />);
    expect(wrapper.text()).toBe('An error occurred!');
  });

  it('snapshotと比較', () => {
    const tree = renderer
      .create(<ErrorMessage message="An error occurred!" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
