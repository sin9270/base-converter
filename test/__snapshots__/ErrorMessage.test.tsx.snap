// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ErrorMessage snapshotと比較 1`] = `
<span
  className="errorMassage"
>
  An error occurred!
</span>
`;
